torch/_inductor/codegen/cpp.py

```diff
@@ -336,7 +336,8 @@ def square(a):
 
     @staticmethod
     def where(a, b, c):
-        return f"decltype({b})::blendv({c}, {b}, {a})"
+        # return a ? b : c
+        return f"decltype({b})::blendv({c}, {b}, {a} > decltype({a})(0))"
```

Comment:
`a` is the condition, why does it require `a` to be greater than `0`?

Reply:
It is tricky. Why is `a` always 0f or 1f? `where` is a common operator. We should not add such limitation or assumption.

Reply:
Is it possible that `a` is already a condition expression and an additional `a>0` check is not necessary?

Reply:
I don't think we will meet a is already a bit condition expression, I think the where's scalar definition can make sure it.

Reply:
btw, it would be `a != 0` instead of `a > 0`?

Reply:
Also, better to do `({a}) > decltype({a})(0)`.
I don't have strong opinion on whether we should explicitly handle the case when `a` is a condition but I think it might be more efficient to do so.